securedrop_client/storage.py

```diff
@@ -177,18 +203,17 @@ def update_submissions(remote_submissions, local_submissions, session, data_dir)
             # A new submission to be added to the database.
             _, source_uuid = submission.source_url.rsplit('/', 1)
             source = session.query(Source).filter_by(uuid=source_uuid)[0]
-            ns = Submission(source=source, uuid=submission.uuid,
-                            size=submission.size,
-                            filename=submission.filename,
-                            download_url=submission.download_url)
+            ns = model(source_id=source.id, uuid=submission.uuid, size=submission.size,
+                       filename=submission.filename, download_url=submission.download_url)
             session.add(ns)
             logger.debug('Added new submission {}'.format(submission.uuid))
 
     # The uuids remaining in local_uuids do not exist on the remote server, so
     # delete the related records.
     for deleted_submission in [s for s in local_submissions
                                if s.uuid in local_uuids]:
-        delete_single_submission_or_reply_on_disk(deleted_submission, data_dir)
+        if is_file:
+            delete_single_submission_or_reply_on_disk(deleted_submission, data_dir)
```

Comment:
Just noting for lower in this file (can't comment on that since it's not in the diff), we should also rewrite `find_new_submissions` -> `find_new_messages` and get rid of the filtering on filename. 

For `mark_file_as_downloaded` we could write a model-independent function for files, messages, replies (since the only think that is different between these three is the type of model object)

Reply:
I left these as separate implementations for now because that's just how we're doing it. I think some of these helper functions only get called once, so it might be better to just define them as local functions since they're passed as callbacks. Because of mocking, this might not be possible, but I think we're overmocking and this change could be part of #263